### PR TITLE
adding monitor for quay operator deployment

### PR DESCRIPTION
**Pull request by eifrach**
Signed-off-by: Eran Ifrach <eifrach@redhat.com>

# Description

Adding monitor for quay deployment instead of waiting 30 sec
in some cases, deployment will take more then 30 sec 

Fixes # (issue)

Error where the pipeline continues before the quay is ready 
output error example 
```conoles
[deploy-disconnected-registry : deploy-disconnected-registry] >> Waiting for the registry deployment to be ready
[deploy-disconnected-registry : deploy-disconnected-registry] I need a resource name to check, value passed: ""
```

## Testing
```console
[deploy-disconnected-registry : deploy-disconnected-registry] >>>> Waiting for ODF Cluster Ready
[deploy-disconnected-registry : deploy-disconnected-registry] >>>>>>>>>>>>>>>>>>>>>>>>>>>>>
[deploy-disconnected-registry : deploy-disconnected-registry] >> Creating the registry deployment
[deploy-disconnected-registry : deploy-disconnected-registry] namespace/ztpfw-registry created
[deploy-disconnected-registry : deploy-disconnected-registry] operatorgroup.operators.coreos.com/quay-operator-operatorgroup created
[deploy-disconnected-registry : deploy-disconnected-registry] subscription.operators.coreos.com/quay-operator created
[deploy-disconnected-registry : deploy-disconnected-registry] INFO: waiting for Quay Operator to be ready ( 3 min )
[deploy-disconnected-registry : deploy-disconnected-registry] >> Waiting for the registry deployment to be ready
[deploy-disconnected-registry : deploy-disconnected-registry] >>>> Checking Resource: deployment with name quay-operator.v3.7.5
[deploy-disconnected-registry : deploy-disconnected-registry] >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
[deploy-disconnected-registry : deploy-disconnected-registry] >> Creating the config for the registry
[deploy-disconnected-registry : deploy-disconnected-registry] secret/config-bundle-secret created
[deploy-disconnected-registry : deploy-disconnected-registry] >> Creating the registry Quay CR
[deploy-disconnected-registry : deploy-disconnected-registry] quayregistry.quay.redhat.com/ztpfw-registry created
```

### CODE REVIEW

File: deploy-disconnected-registry/deploy.sh [hunk 1]
```diff
@@ -146,7 +146,20 @@ function deploy_registry() {
         # Create the registry deployment and wait for it
         echo ">> Creating the registry deployment"
         oc --kubeconfig=${TARGET_KUBECONFIG} -n ${REGISTRY} apply -f ${QUAY_MANIFESTS}/quay-operator.yaml
-        sleep 30
+        QUAY_STATUS=false
+		echo "INFO: waiting for Quay Operator to be ready ( 1 min )"
```
Comment (iranzo) [inline deploy-disconnected-registry/deploy.sh:150]: mixed tabs and spaces for identation

**Review by iranzo — CHANGES_REQUESTED** [review]

**Review by iranzo — APPROVED** [review]
lgtm

File: deploy-disconnected-registry/deploy.sh [hunk 2]
```diff
@@ -146,7 +146,20 @@ function deploy_registry() {
         # Create the registry deployment and wait for it
         echo ">> Creating the registry deployment"
         oc --kubeconfig=${TARGET_KUBECONFIG} -n ${REGISTRY} apply -f ${QUAY_MANIFESTS}/quay-operator.yaml
-        sleep 30
+		QUAY_STATUS=false
+		echo "INFO: waiting for Quay Operator to be ready ( 1 min )"
```
Comment (flaper87) [inline deploy-disconnected-registry/deploy.sh:150]: 1min sounds like too little. Can we do a 2min or 3 max?

**Review by alknopfler — COMMENTED** [review]
LGTM let's test it ok?

**Review by flaper87 — APPROVED** [review]

**Pull request merged**